### ISSUE TITLE
Add coordinated AP for hard mappings 

### ISSUE THREAD

**Issue by pbuttigieg** [ESIPFed/sweet#126]
When classes are "hard mapped " through human expert review between resources like ENVO and SWEET, we need an annotation property that reflects the nature of this link.

Ideally, this would be the same AP across resources

@cmungall any existing AP comes to mind?

**Comment by dr-shorthair** [comment]
Hmm - we are managing a few (very large) linksets but keep them in separate files/graphs to either the source or target. That's why I started the same kind of thing here https://github.com/ESIPFed/sweet/tree/master/alignments 

In fact, we have even gone as far as reifying each mapping so you can attach individual metadata/provenance about each mapping - particularly how/when/who.

**Comment by dr-shorthair** [comment]
Potential approach: 
https://github.com/ESIPFed/sweet/blob/ISSUE-107/alignments/sweet-ssn-mapping.ttl#L78 
(or could even use PROV-O ...)

**Comment by pbuttigieg** [comment]
Hi @dr-shorthair 

Thanks for the input

https://github.com/ESIPFed/sweet/blob/4a5378cb048b6f493cfe57fdb85cf4c90f30baa4/alignments/sweet-dcat-mapping.ttl#L134-L135

So in this model, the owl:equivalentClass is used? 

My understanding of the [OWL documentation](https://www.w3.org/TR/owl-ref/#equivalentClass-def) is that this is used for shared instances. It doesn't actually indicate class equality, which is what we're going for here. owl:sameAs is suggested, but needs OWL Full to be interpreted - not sure where to go from there. 

Looking at the link above:
https://github.com/ESIPFed/sweet/blob/c28c5109bf0a1ff75a5e030a1afa630f2069a414/alignments/sweet-ssn-mapping.ttl#L76-L86

Are you pointing to the provenance code? 
I'm not sure if provenance is exactly right - it's not that one comes from another (necessarily), just that they are aligned.

**Comment by cmungall** [comment]
I would separate the semantics of the mapping from provenance/evidence/confidence of the mapping. Use the appropriate predicate for the former (logical: OWL; looser: SKOS), and use axiom annotation or distinct NGs for the latter.

(I am less keen on using NGs as an alternative to reification due to the necessity of a quad format to communicate multiple triples with different provenance in the same document... and this all hopelessly fails if you try and use OWL.... OWL complicates things further by introducing a new reification vocabulary, sigh).

For the choice of predicate itself it will be either an owl pred or a skos one. This is not the OBO community so you won't be using hasDbXref, which is good. I wrote about these 3 choices here: https://douroucouli.wordpress.com/2019/05/27/never-mind-the-logix-taming-the-semantic-anarchy-of-mappings-in-ontologie/ -- you can ignore stuff about xrefs but the skos vs owl is informative for you.

**Comment by cmungall** [comment]
> My understanding of the OWL documentation is that this is used for shared instances. It doesn't actually indicate class equality, which is what we're going for here. owl:sameAs is suggested, but needs OWL Full to be interpreted - not sure where to go from there

Warning: rabbit holes ahead, watch your step

TL;DR - use owl:equivalentClass if two classes mean exactly the same thing. Do not use owl:sameAs for classes

The OWL documentation is confusing because it is written by formal logicians and they are very careful not to conflate *class* with *class extension*. equivalentClasses states that the extensions (i.e. members) are the same, which is exactly what you want. You don't actually want to state the classes themselves are the same.

Article from Phill's blog on class vs class extension here
http://ontogenesis.knowledgeblog.org/1004

This is a fairly high level description of something that is sort of interesting and goes back to barbers shaving themselves... but really it's irrelevant. If you think two class URIs mean the same thing use equivalentClass

**Comment by cmungall** [comment]
> Are you pointing to the provenance code?

Yep, that's a reified statement

It could be written much more compactly in RDF*

```
sosa:Result subClassOf res;Result << dcterms:created "2017-10-25"^^xsd:date ; .... >>
```

@pbuttigieg note we have 1000s of reified statements in ENVO, but by necessity we use the OWL reification vocabulary.

I wish the semweb community just came up with something like RDF* in the first place to avoid this confusion

My thoughts on RDF*: https://douroucouli.wordpress.com/2019/07/11/proposed-strategy-for-semantics-in-rdf-and-property-graphs/

**Comment by pbuttigieg** [comment]
Thanks for the input @cmungall 

@all do we have a consensus emerging?

**Comment by dr-shorthair** [comment]
I moved the example to a new branch see #129, to separate the concern from #107

**Comment by dr-shorthair** [comment]
Maybe `sosa:usedProcedure` instead of `dct:provenance` ...

**Comment by rduerr** [comment]
I rather like sosa:usedProcedure but I don't like that "Manual assertion" is just a rdfs:comment - this really should be a controlled vocabulary of Procedures....  I can imagine wanting to use the procedure to make decisions in software.

**Comment by dr-shorthair** [comment]
>  I don't like that "Manual assertion" is just a rdfs:comment - this really should be a controlled vocabulary of Procedures

Certainly. I just showed a blank-node with a comment as a placeholder for a names resource (i.e. denoted with a URI).

**Comment by rduerr** [comment]
OK, great!  I suspect the list will be long and sourced from several existing "methods"/"procedure" lists that already exist.
